lib/rubycritic/cli/application.rb

```diff
@@ -14,12 +14,29 @@ def initialize(argv)
 
       def execute
         parsed_options = @options.parse
-        ::Rubycritic::CommandFactory.create(parsed_options).execute
+        result = ::Rubycritic::CommandFactory.create(parsed_options).execute
+
+        return STATUS_ERROR unless satisfy_minimum_score?(result, parsed_options)
+
         STATUS_SUCCESS
       rescue OptionParser::InvalidOption => error
         $stderr.puts "Error: #{error}"
         STATUS_ERROR
       end
+
+      private
+
+      def satisfy_minimum_score?(result, options)
+        return true unless result.is_a?(Hash) && result[:score]
+
+        status = result[:score] >= options.minimum_score
+        if status
+          $stdout.puts "Score: #{result[:score]}"
+        else
+          $stderr.puts "Score (#{result[:score]}) is below the minimum #{options.minimum_score}"
+        end
```

Comment:
The method is a predicate method. Problem is, that you're doing quite a few other things in here. Callers of this method would expect a boolean value back, but not multiple side effects like printing to stdout and so on.